### PR TITLE
Fix logic to determine master vs PR

### PR DESCRIPTION
**Pull request by driazati**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #65157
* **#65155**

This was bugged before on empty strings which caused the hook to write on any job, not just `master` regardless of the `only_on_master` flag.

Test plan: see `[scribe] Skipping RDS write on PR` in the logs for `linux-xenial-cuda11.3-py3.6-gcc7`

Differential Revision: [D31029048](https://our.internmc.facebook.com/intern/diff/D31029048)

### CODE REVIEW

**Comment by driazati** [comment]
@driazati has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31029048).

File: tools/stats/scribe.py [hunk 1]
```diff
@@ -24,9 +27,7 @@ def aws_lambda() -> Any:
 
 
 def invoke_lambda(name: str, payload: Any) -> Any:
-    res = aws_lambda().invoke(
-        FunctionName=name, Payload=json.dumps(payload).encode()
-    )
+    res = aws_lambda().invoke(FunctionName=name, Payload=json.dumps(payload).encode())
```
Comment (malfet) [inline tools/stats/scribe.py:30]: Q: Can we somehow avoid change those back-n-forth?

**Review by malfet — APPROVED** [review]

**Comment by driazati** [comment]
@driazati has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31029048).

File: .github/templates/linux_ci_workflow.yml.j2 [hunk 2]
```diff
@@ -171,6 +171,7 @@ jobs:
           CIRCLE_SHA1: ${{ github.event.pull_request.head.sha || github.sha }}
           CIRCLE_TAG: ${{ steps.parse-ref.outputs.tag }}
           CIRCLE_WORKFLOW_ID: '${{ github.run_id }}_${{ github.run_number }}'
+          GITHUB_WORKFLOW_RUN_ID: '${{ github.run_id }}'
```
Comment (seemethere) [inline .github/templates/linux_ci_workflow.yml.j2:174]: Isn't this already a thing?

`GITHUB_RUN_ID` should give you the same info and it's already built in 

https://docs.github.com/en/actions/reference/environment-variables#default-environment-variables

**Pull request closed without merging**